### DISCUSSION TITLE
how to debug segmentation faults?

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#921] [Q&A] [answered]
With the newest GDL from 14.04.2021 I get sometimes segmentation fault on my Tumbleweed 5.11.11:

Magick: abort due to signal 11 (SIGSEGV) "Segmentation Fault"...

What is the best way to debug this?

**Comment by slayoo** [comment, marked as answer]
One way is to compile with debugging symbols (cmake -DCMAKE_BUILD_TYPE=Debug) and run using ``gdb`` (i.e. ``$ gdb gdl``) where ``gdl`` is the executable file. Then, within ``gdb`` you type ``run`` and after the segfault occurs you can get a stack trace by typing ``bt``. HTH. Thanks!
